sdk/core/azure_core/src/task/standard_spawn.rs

```diff
@@ -27,16 +27,16 @@ impl TaskSpawner for StdSpawner {
         });
 
         // Return a handle that will await the result
-        SpawnHandle(th)
+        SpawnHandle::Std(StdSpawnHandle(th))
     }
 }
 
 #[derive(Debug)]
-pub struct SpawnHandle(std::thread::JoinHandle<()>);
+pub struct StdSpawnHandle(std::thread::JoinHandle<()>);
 
-impl SpawnHandle {
+impl StdSpawnHandle {
     /// Wait for the task to complete and return the result.
-    pub async fn await_result(self) -> crate::Result<()> {
+    pub(crate) async fn wait(self) -> crate::Result<()> {
```

Comment:
Any reason not to just impl `Future`? Did we already talk about that? `wait` is really doing nothing more than polling would with `await` anyway. So it'd be just `handle.await`, which is in line with other `std::thread` and `tokio` tasks.

Reply:
Let me see what that would look like (returning the future directly). There may be some interesting challenges related to the different return types, but I may be able to make this work.